lib/optimizely.rb

```diff
@@ -378,7 +378,7 @@ def get_feature_variable_for_type(feature_flag_key, variable_key, variable_type,
       # attributes - Hash representing visitor attributes and values which need to be recorded.
       #
       # Returns the type-casted variable value.
-      # Returns nil if the feature flag or variable are not found.
+      # Returns nil if the feature flag, variable, User ID are not found and variable type differs.
```

Comment:
Please replace with below line and make a PR to Optimizely. 
Returns nil if the feature flag or variable or user ID is empty 
                  in case of variable type mismatch